templates/operations.ts.hbs

```diff
@@ -187,26 +187,24 @@
         headers[USER_AGENT_HEADER] = [headers[USER_AGENT_HEADER], USER_AGENT_VALUE].join(" ");
       }
 
-      const requestOptions = {
-        ...this.clientConfig.fetchOptions,
-        {{#if (isRequestWithPayload request)}}body: this.clientConfig.transformRequest(options.body, headers),{{/if}}
-        headers,
-        method: "{{loud method}}"
-      };
+      const response = await doFetch(
+        url.toString(),
+        {
+          method: "{{loud method}}",
+          headers,
+          {{#if (isRequestWithPayload request)}}body: this.clientConfig.transformRequest(options.body, headers){{/if}}
+        }, 
+        this.clientConfig,
+        rawResponse
+      )
 
-      const response = await fetch(url.toString(), requestOptions);
+      {{#if (eq (getReturnTypeFromOperation this) "void") }}
       if (rawResponse) {
-        return response;
-      } else if (this.clientConfig.throwOnBadResponse && !response.ok && response.status !== 304) {
-        throw new ResponseError(response);
+        return response as Response;
       }
-      {{#unless (eq (getReturnTypeFromOperation this) "void") }}
-        else {
-          const text = await response.text();
-          // It's ideal to get "{}" for an empty response body, but we won't throw if it's truly empty
-          return text ? JSON.parse(text) : {};
-        }
-      {{/unless}}
+      {{else}}
+      return response as Response | {{getReturnTypeFromOperation this}};
```

Comment:
Can you explain this line a little bit more for me. If we are expecting that we aren't using a _rawResponse_ we are saying that the return value is the **union** of a Response object and the return type of this operation? Why wouldn't it simply be the return type of the operation without the union?

Reply:
We use the `rawResponse` when we call the `doFetch` helper function for the `response` variable. Since we don't know what `rawResponse` is when we pass it to `doFetch`, the result is either a response object OR the return type

Reply:
I think it is to handle any unexpected response - the response that is not defined in RAML specially in case of error response